Various github-post and issues cleanup

benesch opened the pull request:
See commit messages for details.

benesch committed:
github-post: don't post about subtests

When subtests fail, don't post issues about them to reduce issue spam.
Just let the parent test failure serve as the umbrella issue.

Fix #27749.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/27836)
<!-- Reviewable:end -->

benesch committed:
issues: don't prepend cockroach package prefix

Since switching to 'go tool test2json', the cockroach package prefix
(github.com/cockroachdb/cockroach/pkg) is already included in the
package name that is provided to github-post. Teach the issues package
not to prefix it *again* with the cockroach package prefix. The goal is
a short issue title that is free of the cockroach package prefix.

Once this patch lands, I'll strip the prefixes from the titles of any
issues that stress jobs have posted recently.

Release note: None

benesch committed:
issues: remove unnecessary function parameter

Release note: None

petermattis reviewed (approved):
<img class="emoji" title=":LGTM:" alt=":LGTM:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 3 of 3 files at r1, 2 of 2 files at r2, 1 of 1 files at r3.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/27836#-:-LI6i4E4AiZDSDLPzpYx:bmxilld)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale)



<!-- Sent from Reviewable.io -->

benesch commented:
bors r=petermattis
